src/core/server/http/http_server.ts

```diff
@@ -246,6 +260,18 @@ export class HttpServer {
     }
   }
 
+  private setupGracefulShutdownHandlers() {
+    this.registerOnPreRouting((request, response, toolkit) => {
+      if (this.stopped) {
```

Comment:
Do we need to dissociate the `stopping` and `stopped` states or are we good just using a single flag?

Reply:
That's a great question. Do we have any use cases for that? After looking at the other places where we're using this flag, they are just `log.warn`-ing when a setup/start method is called after calling `stop`. 

I could rename `this.stopped` to `this.stopping` and let future use cases to actually implement the `stopped` flag. Does it make sense?

Reply:
> I could rename this.stopped to this.stopping and

It can be confusing that plugin has `stopping` status after it has been stopped. It may make sense to separate phases as long as we support `async stop`.

Reply:
I've created the new 2 flags: `stopping` and `stopped`